tests/test_bkr_handlers_mocked.py

```diff
@@ -0,0 +1,456 @@
+        elif new_state == 1:
+            self._append(lines=a.par1)
+        elif new_state == 2:
+            self._append(lines=a.par2)
+        elif new_state == 3:
+            self._append(lines=a.par3)
+        elif new_state == 4:
+            self._append(lines=a.footer)
+
+
+class ChangingFile(LivingFile):
+    """
+    Middle section of the file is changing once per tick.
+    """
+
+    name = 'changing'
+
+    def _tick(self, new_state):
+        a = self.make_autotext()
+        if new_state == 0:
+            self._write(lines=a.header+a.par0+a.footer)
+        elif new_state == 1:
+            self._write(lines=a.header+a.par1+a.footer)
+        elif new_state == 2:
+            self._write(lines=a.header+a.par2+a.footer)
+        elif new_state == 3:
+            self._write(lines=a.header+a.par3+a.footer)
+
+
+class ShrinkingFile(LivingFile):
+
+    name = 'shrinking'
+
+    def _tick(self, new_state):
+        a = self.make_autotext()
+        if new_state == 0:
+            self._write(lines=a.header+a.par1+a.par2+a.par3+a.footer)
+        elif new_state == 1:
+            self._write(lines=a.header+a.par1+a.par2+a.footer)
+        elif new_state == 2:
+            self._write(lines=a.header+a.par1+a.footer)
+        elif new_state == 3:
+            self._write(lines=a.header+a.footer)
+
+
+class ResultTracker:
+    """
+    Utility class to keep track of mocked SUT http_put() function
+    and help collect data relevant for further test asserts.
+    """
+
+    def __init__(self, mock, file):
+        """
+        Create tracker of http_put Mock object *mock*, tracking
+        calls that upload files represented by LivingFile *file*.
+        """
+        self.file = file
+        self.mock = mock
+
+    def debug_stats(self):
+        """
+        Describe what our http_put() and LivingFile have been up to.
+        """
+        def describe_call(C):
+            url = C[0][0]
+            payload = C[0][1]
+            return (
+                "call sending %d bytes to http:/..%s, preview: %r .. %r"
+                % (
+                    len(payload),
+                    url[-20:],
+                    payload[:20].decode(),
+                    payload[-20:].decode(),
+                )
+            )
+        debug_var(self, 'debug_stats', 'self.file.wrote_so_far', self.file.wrote_so_far)
+        debug_var(self, 'debug_stats', 'self.file.expected_size', self.file.expected_size)
+        debug_var(self, 'debug_stats', 'len(self.calls)', len(self.calls))
+        for call in self.calls:
+            debug_msg(self, 'debug_stats', ' .. ' + describe_call(call))
+
+    @property
+    def calls(self):
+        """
+        Return all calls from tracked object (Mock.call_args_list)
+        where url contains name of tracked file.
+        """
+        return [c for c in self.mock.call_args_list
+                if self.file.name in c[0][0]]
+
+    @property
+    def ttl_uploaded(self):
```

Comment:
Please don't use "ttl" as it's abbreviation for "Time To Live" which is directly related to networking and is confusing when used as "total" together with "uploaded".

Reply:
renamed to "total"